src/flamenco/runtime/tests/fd_exec_instr_test.c

```diff
@@ -896,36 +896,23 @@ _txn_context_create_and_exec( fd_exec_instr_test_runner_t *      runner,
   tpool->worker_cnt = 1;
   tpool->worker_max = 1;
 
-  int res = fd_runtime_prepare_txns_phase1( slot_ctx, task_info, txn, 1 );
-  (void)res;
-  // if (res != 0) {
-  //   return task_info;
-  // }
-
-  res |= fd_runtime_prepare_txns_phase2_tpool( slot_ctx, task_info, 1, tpool );
-  // if (res != 0) {
-  //   return task_info;
-  // }
-
-  res |= fd_runtime_prepare_txns_phase3( slot_ctx, task_info, 1 );
-  // if (res != 0) {
-  //   return task_info;
-  // }
-
-  // Below is a stripped-out version of fd_runtime_execute_txn_task because the Agave harness does not reclaim accounts
-  if( !( task_info->txn->flags & FD_TXN_P_FLAGS_SANITIZE_SUCCESS ) ) {
-    // At this point the transaction error code should have been propogated
-    // to task_info->exec_res
-    return task_info;
-  }
-
-  res = fd_execute_txn_prepare_phase4( task_info->txn_ctx );
-  // if( res != 0 ) {
-  //   return task_info;
-  // }
-  task_info->txn->flags |= FD_TXN_P_FLAGS_EXECUTE_SUCCESS;
-
-  task_info->exec_res = fd_execute_txn( task_info->txn_ctx );
+  fd_runtime_prepare_txns_start( slot_ctx, task_info, txn, 1UL );
+
+  fd_runtime_pre_execute_check( task_info );
+
+  if( !task_info->exec_res ) {
+    int res = fd_execute_txn_prepare_finish( task_info->txn_ctx );
+    if( FD_UNLIKELY( res ) ) {
+      FD_LOG_ERR(("could not prepare txn"));
+    }
+
+    task_info->txn->flags |= FD_TXN_P_FLAGS_EXECUTE_SUCCESS;
+    task_info->exec_res    = fd_execute_txn( task_info->txn_ctx );
+  }
+
+  slot_ctx->slot_bank.collected_execution_fees += task_info->txn_ctx->execution_fee;
+  slot_ctx->slot_bank.collected_priority_fees  += task_info->txn_ctx->priority_fee;
```

Comment:
Instead of doing this, can you change `txn_result->fee_details.transaction_fee` and `txn_result->fee_details.prioritization_fee` to use the fees from the txn_ctx?

Reply:
keeping it consistent with the actual existing code path used by replay so going to keep it like this